### PR TITLE
Add line styles to TextGrid

### PR DESCRIPTION
**Pull request by andydotxyz**
### Description:
As well as cell styles we want to support styles for rows (that extend into empty cells too)
This is a breaking API change, but none of it was released befpre

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.
- [x] Public APIs match existing style.
- [ ] Any breaking changes have a deprecation path or have been discussed. <- sorry, I didn't check in advance

### CODE REVIEW

**Comment by andydotxyz** [comment]
Preview:

Screenshot from 2020-04-29 21-52-52

File: cmd/fyne_demo/screens/widget.go [hunk 1]
```diff
@@ -30,11 +30,21 @@ func makeButtonTab() fyne.Widget {
...
+	grid := widget.NewTextGridFromString("TextGrid\n  Content\nZebra")
 	grid.SetStyleRange(0, 0, 0, 3,
 		&widget.CustomTextGridStyle{FGColor: color.RGBA{R: 0, G: 0, B: 128, A: 255}})
 	grid.SetStyleRange(0, 4, 0, 7,
 		&widget.CustomTextGridStyle{BGColor: &color.RGBA{R: 128, G: 0, B: 0, A: 255}})
+	grid.Content[1].Style = &widget.CustomTextGridStyle{BGColor: &color.RGBA{R: 64, G: 64, B: 64, A: 128}}
+
+	white := &widget.CustomTextGridStyle{FGColor: &color.RGBA{R: 255, G: 255, B: 255, A: 255}}
+	black := &widget.CustomTextGridStyle{FGColor: &color.RGBA{R: 0, G: 0, B: 0, A: 255}}
+	grid.Content[2].Content[0].Style = white
```
Comment (stuartmscott) [inline cmd/fyne_demo/screens/widget.go:42]: Content Content is a bit confusing
```suggestion
	grid.Row[2].Cell[0].Style = white
```

Comment (andydotxyz) [reply cmd/fyne_demo/screens/widget.go:42]: Yes, I could not decide if consistency worked here.
Do you have a good suggestion? I originally had "Cells" but it seemed a little uninspired...

Comment (andydotxyz) [reply cmd/fyne_demo/screens/widget.go:42]: Oh, haha I see your suggestion now sorry. It’s cunning. But if the index were row and col then perhaps plurals read better? Grid.Rows[row].Cells[col]?

**Comment by andydotxyz** [comment]
Pushed the new name, thanks

**Pull request merged**